contracts/integrations/test/actors/pool_operator.ts

```diff
@@ -33,6 +37,17 @@ export function PoolOperatorMixin<TBase extends Constructor>(Base: TBase): TBase
             // tslint:disable-next-line:no-inferred-empty-object-type
             super(...args);
             this.actor = (this as any) as Actor;
+
+            // Register this mixin's assertion generators
+            if (this.actor.simulationEnvironment !== undefined) {
```

Comment:
Should it construct a simulation if one is undefined?

Reply:
Or at least raise an exception so we don't fail silently.